### ISSUE TITLE
🚨 Error: Cannot read property 'getBoundingClientRect' of undefined

### ISSUE THREAD

**Issue by ampprojectbot** [ampproject/amphtml#31898]
Details
---
**Error report:** [link](http://go/ampe/CMvVkLjoh5HKdA)
**First seen:** Nov 19, 2020
**Frequency:** ~ 945/day

Stacktrace
---
<pre><code>Error: Cannot read property 'getBoundingClientRect' of undefined
    at getBoundingClientRect (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/extensions/amp-base-carousel/0.1/dimensions.js#L58">extensions/amp-base-carousel/0.1/dimensions.js:58</a>:16)
    at getDimension (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/extensions/amp-base-carousel/0.1/dimensions.js#L73">extensions/amp-base-carousel/0.1/dimensions.js:73</a>:23)
    at getCenter (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/extensions/amp-base-carousel/0.1/dimensions.js#L97">extensions/amp-base-carousel/0.1/dimensions.js:97</a>:6)
    at getPosition (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/extensions/amp-base-carousel/0.1/dimensions.js#L155">extensions/amp-base-carousel/0.1/dimensions.js:155</a>:16)
    at getPercentageOffsetFromAlignment (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/extensions/amp-base-carousel/0.1/carousel.js#L1241">extensions/amp-base-carousel/0.1/carousel.js:1241</a>:25)
    at restoreScrollStart_ (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/extensions/amp-base-carousel/0.1/carousel.js#L1218">extensions/amp-base-carousel/0.1/carousel.js:1218</a>:6)
    at mutator (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/src/service/mutator-impl.js#L227">src/service/mutator-impl.js:227</a>:8)
    at callback (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/src/service/vsync-impl.js#L470">src/service/vsync-impl.js:470</a>:16)
    at callTask_ (<a href="https://github.com/ampproject/amphtml/blob/2012301722000/src/service/vsync-impl.js#L426">src/service/vsync-impl.js:426</a>:8)
</code></pre>

Notes
---
`@rsimha` modified `extensions/amp-base-carousel/0.1/dimensions.js:58-58` in #21212 (May 16, 2019)
`Sepand Parhami` modified `extensions/amp-base-carousel/0.1/dimensions.js:67-86` in #20389 (Jan 17, 2019)
`Sepand Parhami` modified `extensions/amp-base-carousel/0.1/carousel.js:1216-1229` in #20293 (Jan 16, 2019)
`@powerivq` modified `src/service/mutator-impl.js:225-227` in #26638 (Feb 12, 2020)
`@jridgewell` modified `src/service/vsync-impl.js:470-471` in #20836 (Feb 14, 2019)

**Seen in:**
- 11-25 Stable (2111+3)
- 12-23 Control (2217)
- 12-23 Stable (2217)
- 12-30 Experiment-A (1722)
- +6 more

**Possible assignees:** `@powerivq`, `@jridgewell`

/cc @ampproject/release-on-duty

**Comment by micajuine-ho** [comment]
Took a quick peak at the error links--all of the carousels on those pages seem to have the same structure:
```
<p>
  <amp-carousel>
    <div class="ampforwp-gallery-item amp-carousel-container"><amp-img></amp-img></div>
    <div class="ampforwp-gallery-item amp-carousel-container"><amp-img></amp-img></div>
    <div class="ampforwp-gallery-item amp-carousel-container"><amp-img></amp-img></div>
  </amp-carousel>
</p>
```

`<p>` tags are only allowed to have "paragraph content" within them (all other tags, including divs, will get stripped by the browser). This causes all the carousels to have no slides--which probably causes the error found above.

This is essentially the same problem as https://github.com/ampproject/amphtml/issues/31251.

Maybe this sort of developer error can be fixed with wider documentation for 'common issues'?

/cc @ampproject/wg-components

**Comment by micajuine-ho** [comment]
Closed via #31945
